Features/Inventory/BigInventory.cs

```diff
@@ -0,0 +1,365 @@
+			if (remainingStack == null || remainingStack.Count > 1)
+			{
+				_logger.Error("Single AddItem return more than one stack");
+				return remainders;
+			}
+
+			// Append dictionary entry (1!) to overflow
+			if (remainingStack.Count > 0)
+			{
+				remainders.Add(index, remainingStack[0]);
+			}
+		}
+
+		return remainders;
+	}
+
+	public List<ItemStack> GetContents()
+	{
+		return _inventories.Values.SelectMany(inventory => inventory.GetContents()).ToList();
+	}
+
+	public void SetContents(List<ItemStack> items)
+	{
+		if (items.Count > Size)
+		{
+			_logger.Error("Too many items to fit in inventory");
+			return;
+		}
+
+		var index = 0;
+		foreach (var inventory in _inventories.Values)
+		{
+			var sourceItems = items.GetRange(index, inventory.Size);
+			inventory.SetContents(sourceItems);
+			index += inventory.Size;
+		}
+	}
+
+	public bool Contains(string itemId)
+	{
+		return _inventories.Values.Any(inventory => inventory.Contains(itemId));
+	}
+
+	public bool Contains(ItemStack item)
+	{
+		return _inventories[item.Category].Contains(item);
+	}
+
+	public bool Contains(string itemId, int amount)
+	{
+		return _inventories.Values.Any(inventory => inventory.Contains(itemId, amount));
+	}
+
+	public bool Contains(ItemStack item, int amount)
+	{
+		return _inventories[item.Category].Contains(item, amount);
+	}
+
+	public bool ContainsAtLeast(ItemStack item, int amount)
+	{
+		return _inventories[item.Category].ContainsAtLeast(item, amount);
+	}
+
+	public Dictionary<int, ItemStack> All(string itemId)
+	{
+		var all = new Dictionary<int, ItemStack>();
+		var index = 0;
+		foreach (var inventory in _inventories.Values)
+		{
+			var items = inventory.All(itemId);
+			foreach (var (key, value) in items)
+			{
+				all.Add(key + index, value);
+			}
+
+			index += inventory.Size;
+		}
+
+		return all;
+	}
+
+	public Dictionary<int, ItemStack> All(ItemStack item)
+	{
+		var indexOffset = 0;
+		var results = new Dictionary<int, ItemStack>();
+		foreach (var (category, inventory) in _inventories)
+		{
+			// Technically redundant since wrongInventory.First() will return empty dictionary
+			if (category != item.Category)
+			{
+				indexOffset += inventory.Size;
+			}
+
+			// Get local indices and add offset for "global" index
+			var indices = inventory.All(item);
+			foreach (var (stackIndex, stack) in indices)
+			{
+				results.Add(stackIndex + indexOffset, stack);
+			}
+		}
+
+		return results;
+	}
+
+	public int First(string itemId)
+	{
+		var indexOffset = 0;
+		foreach (var inventory in _inventories.Values)
+		{
+			var first = inventory.First(itemId);
+			if (first != -1)
+			{
+				return first + indexOffset;
+			}
+
+			indexOffset += inventory.Size;
+		}
+
+		return -1;
+	}
+
+	public int First(ItemStack item)
+	{
+		var indexOffset = 0;
+		foreach (var (category, inventory) in _inventories)
+		{
+			// Technically redundant since wrongInventory.First() will return -1
+			if (category != item.Category)
```

Comment:
Added `continue` to the condition in `All(ItemStack)`

On another note, i do not know what 
`// Technically redundant since wrongInventory.First() will return empty dictionary`
is refering to. I will assume past person does know better. Will keep this line